### PR TITLE
Improve docs for MR and PR comments

### PR DESCRIPTION
**Pull request by chmccreery**
# Thanks for improving Semgrep Docs 😀

### Please ensure

- [ ] A subject matter expert (SME) reviews the content
- [ ] A technical writer reviews the content or PR
- [x] This change has no security implications or else you have pinged the security team
- [x] Redirects are added if the PR changes page URLs

### CODE REVIEW

**Comment by s-santillan** [comment]
@pabloest Hi Pablo! If I am going to adopt this, what's the correct way of doing it? I was thinking - (1) close this PR, (2) add my changes to her branch locally, (3) make a new PR (so I will get the notifs for review changes instead of Clara).

Does this work for you?

**Comment by pabloest** [comment]
That works for me. You could also just push to this branch if Clara is ok with that and the notification implications. Both seem fine.

**Comment by s-santillan** [comment]
@pabloest Got it, thank you! @chmccreery How do you prefer to handle this?

**Comment by s-santillan** [comment]
@chmccreery I can't request a review from you (as this is your branch) but here I am commenting for one as you are SME :) 

@pabloest requesting for TW review. It is not my aim to rewrite the whole notif article (though it should probably be rewritten, as there is overlap with Integrations). I can make a separate PR for that improvement down the line.

File: docs/semgrep-app/notifications.md [hunk 1]
```diff
@@ -59,13 +59,13 @@ Automated comments on GitHub pull requests look like this:
 <br />
 An inline GitHub pull request comment.
 
-Note that [Semgrep App](https://semgrep.dev/manage) uses the permissions requested by [the Semgrep GitHub App](https://github.com/marketplace/semgrep-dev) to leave PR comments.
+Note that [Semgrep App](https://semgrep.dev/manage) uses the permissions requested by [the Semgrep GitHub App](https://github.com/marketplace/semgrep-dev) to leave PR comments. You can verify that you have granted these permissions by visiting either https://github.com/organizations/<your_org_name>/settings/installations or https://github.com/organizations/<your_org_name>/<your_repo_name>/settings/installations).
```
Comment (pabloest) [inline docs/semgrep-app/notifications.md:62]:
```suggestion
Note that [Semgrep App](https://semgrep.dev/manage) uses the permissions requested by [the Semgrep GitHub App](https://github.com/marketplace/semgrep-dev) to leave PR comments. You can verify that you have granted these permissions by visiting either `https://github.com/organizations/<your_org_name>/settings/installations` or `https://github.com/organizations/<your_org_name>/<your_repo_name>/settings/installations`.
```

File: docs/semgrep-app/notifications.md [hunk 2]
```diff
@@ -59,13 +59,13 @@ Automated comments on GitHub pull requests look like this:
...
 
-If you are using Github Actions to run Semgrep, no extra changes are needed to get PR comments. If you are using another CI provider, in addition to the environment variables you set after following [sample CI configurations](/semgrep-ci/sample-ci-configs/) you need to ensure that the following environment variables are correctly defined:
+If you are using GitHub Actions to run Semgrep, no extra changes are needed to get PR comments. If you are using another CI provider, in addition to the environment variables you set after following [sample CI configurations](/semgrep-ci/sample-ci-configs/) you need to ensure that the following environment variables are correctly defined:
 
-- `SEMGREP_COMMIT` is set to the full commit hash of the code being scanned (e.g. `d8875d6a63bba2b377a57232e404d2e367dce82d`)
-- `SEMGREP_PR_ID` is set to the PR number of the pull request on Github (e.g. `2900`)
+- `SEMGREP_PR_ID` is set to the PR number of the pull request on Github (e.g. `2901`)
```
Comment (pabloest) [inline docs/semgrep-app/notifications.md:66]:
```suggestion
- `SEMGREP_PR_ID` is set to the PR number of the pull request on Github (for example, `2901`)
```

File: docs/semgrep-app/notifications.md [hunk 3]
```diff
@@ -59,13 +59,13 @@ Automated comments on GitHub pull requests look like this:
...
 - `SEMGREP_REPO_NAME` is set to the repo name (e.g., `returntocorp/semgrep`)
```
Comment (pabloest) [inline docs/semgrep-app/notifications.md:67]:
```suggestion
- `SEMGREP_REPO_NAME` is set to the repo name (for example, `returntocorp/semgrep`)
```

File: docs/semgrep-app/notifications.md [hunk 4]
```diff
@@ -59,13 +59,13 @@ Automated comments on GitHub pull requests look like this:
...
+- `SEMGREP_REPO_URL` is set to the repository URL where your project is viewable online (e.g., `https://github.com/returntocorp/semgrep`)
```
Comment (pabloest) [inline docs/semgrep-app/notifications.md:68]:
```suggestion
- `SEMGREP_REPO_URL` is set to the repository URL where your project is viewable online (for example, `https://github.com/returntocorp/semgrep`)
```

**Review by pabloest — APPROVED** [review]
Very minor suggestions added in line, otherwise this looks good to me.

File: docs/semgrep-app/notifications.md [hunk 5]
```diff
@@ -112,6 +110,35 @@ If you would like to allow Semgrep to automatically suggest quick fixes, visit t
...
+![Screenshot of a sample autofix PR suggestion](../img/notifications-github-suggestions.png)
+
+
+#### Enabling autofix for your GitLab or GitHub code repository
+
+
+**Prerequisites:**
+
+* **For GitLab users:** GitLab users must be signed in to semgrep.dev as they require a Semgrep API token.
+* **For GitHub users:** Ensure you have granted sufficient permissions requested by Semgrep GitHub App for the repository or organization that will receive PR autofix suggestions.
```
Comment (chmccreery) [inline docs/semgrep-app/notifications.md:131]: IMO these really belong in the PR comments and MR comments sections. The only pre-requisite here is that you have followed the instructions in those sections and are successfully receiving inline PR or MR comments. This makes it sound like there are additional pre-requisites for autofix.

File: docs/semgrep-app/notifications.md [hunk 6]
```diff
@@ -112,6 +110,35 @@ If you would like to allow Semgrep to automatically suggest quick fixes, visit t
...
+
+1. Sign in to your [Semgrep App account](https://semgrep.dev/login).
+2. Click **Projects** from the **App sidebar**.
+3. Click the name of the project to enable autofix for.
+4. Click the toggle for **Autofix (beta)**.
+![Screenshot of autofix toggle](../img/notifications-enable-autofix.png)
+5. After enabling autofix, your succeeding scans will include PR or MR suggestions where applicable.
```
Comment (chmccreery) [inline docs/semgrep-app/notifications.md:138]: 5. After enabling autofix, your inline PR or MR comments will include code suggestions for applicable rules

Comment (s-santillan) [reply docs/semgrep-app/notifications.md:138]: @chmccreery Does this mean that RIGHT after enabling autofix in App, the most recent scan will update itself to show these comments?

Comment (chmccreery) [reply docs/semgrep-app/notifications.md:138]: No previously-left comments will be updated, but all comments that are left _after_ autofix was enabled in App will have a suggestion, if the rule that caught the finding has a "fix" field or a "fix-regex" field.

Comment (s-santillan) [reply docs/semgrep-app/notifications.md:138]: How about:

5. All scans performed after enabling autofix will generate inline PR or MR comments with code suggestions for applicable rules?

Comment (s-santillan) [reply docs/semgrep-app/notifications.md]: (Decided to go ahead with the above.)

**Comment by s-santillan** [comment]
@chmccreery Have resolved your last two items. Feel free to squash and merge if they look good to you.

File: docs/semgrep-app/notifications.md [hunk 7]
```diff
@@ -106,7 +106,35 @@ semgrep:
     GITLAB_TOKEN: $PAT
 ```
 
-NOTE: GitLab MR comments are only available to logged-in semgrep.dev users, requiring both a Semgrep deployment ID and a Semgrep API token.
+If you would like to allow Semgrep to automatically suggest quick fixes, visit the [projects page](https://semgrep.dev/orgs/-/projects), select a project, and toggle "Autofix" on.
+
```
Comment (chmccreery) [inline docs/semgrep-app/notifications.md:110]: Is this now redundant with the autofix section below?

**Review by chmccreery — COMMENTED** [review]
The new image for semgrep-pull-request is very obviously me testing things, and not a real find. In fact, the message doesn't even match the line of code. Can we use one of the ones in https://github.com/returntocorp/semgrep-app/pull/4691 instead? (blocking)

File: docs/semgrep-app/notifications.md (hunk 7 above)
Comment (s-santillan) [reply docs/semgrep-app/notifications.md:110]: Good catch. Removing it.

**Comment by s-santillan** [comment]
> The new image for semgrep-pull-request is very obviously me testing things, and not a real find. In fact, the message doesn't even match the line of code. Can we use one of the ones in [returntocorp/semgrep-app#4691](https://github.com/returntocorp/semgrep-app/pull/4691) instead? (blocking)

Gotcha, I hope my next example is better - I chose a rather simple one. Feel free to squash and merge if this is ok.

**Pull request merged**